pkg/alertmanager/amcfg.go

```diff
@@ -808,6 +808,19 @@ func (cg *configGenerator) convertHTTPConfig(ctx context.Context, in monitoringv
 		if username != "" && password != "" {
 			out.BasicAuth = &basicAuth{Username: username, Password: password}
 		}
+	} else if in.Authorization != nil {
```

Comment:
Could we warn when `Authorization` and `BasicAuth` are set?

Reply:
This is available only in alertmanager 0.22+, so I believe we need a version check.

Reply:
Thanks @paulfantom 
Where can I find an example of a version check ?

Also, does the following log suffice:

```go
if in.BasicAuth != nil && in.Authorization != nil {
		level.Warn(cg.logger).Log("msg", "Basic auth and Authorization are mutually exclusive. Basic auth will take precedence.")
	}
111

Reply:
Yes, that message looks good :+1: 

Here is an example for version check we do in statefulset.go: https://github.com/prometheus-operator/prometheus-operator/blob/master/pkg/alertmanager/statefulset.go#L258-L262

Reply:
Thank you :D

Reply:
I added the check in the operator code as this is the only place we have access to the Alertmanager version

Reply:
Side-note: it might make sense to pass the Alertmanager version to `configGenerator`.

Reply:
I can it there as well and not render the authorization part as well